### PR TITLE
`Development`: Update programming exercise ICL docs

### PR DESCRIPTION
**Pull request by BBesrour**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Added a section about editing the checkout paths

### Steps for Testing

Make sure that [docs](https://artemis-platform--9509.org.readthedocs.build/en/9509/user/exercises/programming.html#edit-repositories-checkout-paths) are correct and comprehensible

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Screenshots

image

image

## Summary by CodeRabbit

- **New Features**
	- Introduced a validation bar for improved navigation during programming exercise setup.
	- Added sections for editing repository checkout paths, configuring static code analysis, and verifying exercise configurations.

- **Documentation**
	- Enhanced clarity and usability of the programming exercise setup guide with additional notes and warnings.
	- Removed specific version constraints for Sphinx-related packages to allow for the latest updates.

### CODE REVIEW

**Review by cremertim — DISMISSED** [review]

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
-Sphinx==7.4.7
-sphinx-rtd-theme==2.0.0
-sphinx-autobuild==2024.9.19
-sphinxcontrib-bibtex==2.6.3
+Sphinx
+sphinx-rtd-theme
+sphinx-autobuild
+sphinxcontrib-bibtex
```
Comment (SimonEntholzer) [inline docs/requirements.txt:4]: Any particular reason you removed the version here? Seems reasonable though

**Review by SimonEntholzer — DISMISSED** [review]
Lgtm

File: docs/requirements.txt (hunk 1 above)
Comment (BBesrour) [reply docs/requirements.txt]: Those version didnt work for me. But I reverted since we should rather replace them with the newer versions

**Review by raffifasaro — APPROVED** [review]
Changes LGTM

**Review by SimonEntholzer — DISMISSED** [review]
re-approve after addition of timeout docs

**Review by SimonEntholzer — APPROVED** [review]

**Review by raffifasaro — APPROVED** [review]
re-approval

**Review by HawKhiem — APPROVED** [review]
LGTM

**Pull request merged**